### PR TITLE
All/Any intrinsic lowering

### PR DESCRIPTION
**Pull request by mleair**
This incorporates comments from the  ml-allany branch PR #727

### CODE REVIEW

File: flang/test/Lower/intrinsic-procedures.f90 [hunk 1]
```diff
@@ -44,6 +44,18 @@ subroutine aint_test(a, b)
...
+! ALL
+! CHECK-LABEL: all_test
+! CHECK-SAME: %[[arg0:.*]]: !fir.box<!fir.array<?x!fir.logical<4>>>) -> !fir.logical<4>
+logical function all_test(mask)
+  logical :: mask(:)
+! CHECK: %[[c1:.*]] = constant 1 : index
+! CHECK: %[[a1:.*]] = fir.convert %[[arg0]] : (!fir.box<!fir.array<?x!fir.logical<4>>>) -> !fir.box<none>
+! CHECK: %[[a2:.*]] = fir.convert %[[c1]] : (index) -> i32
+  all_test = all(mask)
+! CHECK:  %[[a3:.*]] = fir.call @_FortranAAll(%[[a1]], %{{.*}}, %[[a2]]) : (!fir.box<none>, !fir.ref<i8>, i32, i32) -> i1
```
Comment (schweitzpgi) [inline flang/test/Lower/intrinsic-procedures.f90:56]: Looks like this is still eating 2 arguments.

**Review by schweitzpgi — APPROVED** [review]
Thank you. Looks good.

File: flang/lib/Lower/IntrinsicCall.cpp [hunk 2]
```diff
@@ -1149,6 +1164,53 @@ mlir::Value IntrinsicLibrary::genAimag(mlir::Type resultType,
...
+       Fortran::lower::createTempMutableBox(builder, loc, resultArrayType);
+  auto resultIrBox =
+       Fortran::lower::getMutableIRBox(builder, loc, resultMutableBox);
+
+  // Call runtime. The runtime is allocating the result.
+  Fortran::lower::genAllDescriptor(builder, loc, resultIrBox, mask, dim);
+  return Fortran::lower::genMutableBoxRead(builder, loc, resultMutableBox)
+      .match(
+          [&](const fir::ArrayBoxValue &box) -> fir::ExtendedValue {
+            return box;
```
Comment (jeanPerier) [inline flang/lib/Lower/IntrinsicCall.cpp:1206]: The clean-up (that insert deallocation (and finalization in the future), after the statement) is still needed here, it looks like you removed it (`addCleanUpForTemp`).

File: flang/lib/Lower/IntrinsicCall.cpp [hunk 3]
```diff
@@ -1173,6 +1235,52 @@ mlir::Value IntrinsicLibrary::genAnint(mlir::Type resultType,
...
+       Fortran::lower::createTempMutableBox(builder, loc, resultArrayType);
+  auto resultIrBox =
+       Fortran::lower::getMutableIRBox(builder, loc, resultMutableBox);
+
+  // Call runtime. The runtime is allocating the result.
+  Fortran::lower::genAnyDescriptor(builder, loc, resultIrBox, mask, dim);
+  return Fortran::lower::genMutableBoxRead(builder, loc, resultMutableBox)
+      .match(
+          [&](const fir::ArrayBoxValue &box) -> fir::ExtendedValue {
+            return box;
```
Comment (jeanPerier) [inline flang/lib/Lower/IntrinsicCall.cpp:1277]: Same here (`addCleanUpForTemp`).

**Pull request merged**